spec/api/v2/admin/api_keys_spec.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+describe 'Api::V2::Admin::APIKeys' do
+  include_context 'bearer authentication'
+
+  let!(:create_member_permission) do
+    create :permission,
+           role: 'member'
+  end
+  let!(:create_admin_permission) do
+    create :permission,
+           role: 'admin'
+  end
+
+  describe 'GET /api/v2/admin/api_keys' do
+    let!(:test_user) { create(:user, otp: otp_enabled) }
+    let(:otp_enabled) { true }
+    let!(:api_key) { create :api_key, user: test_user }
+    let(:expected_fields) do
+      {
+        kid: api_key.kid,
+        state: api_key.state,
+        scope: %w[trade]
+      }
+    end
+    let(:params) do
+      {
+        uid: test_user.uid
+      }
+    end
+    let(:do_request) { get "/api/v2/admin/api_keys", params: params, headers: auth_header }
+
+    context 'successful' do
+      it 'Returns api keys for selected account' do
+        do_request
+        expect(response.status).to eq(200)
+        expect(json_body.first).to include(expected_fields)
```

Comment:
can you ensure the secret is absent from the response here

Reply:
yes it is. But we will need to change the flow once we get merged new Vault policies, as Maksim has changed the way it work for secret key for API Keys.

I will add it in the test, so it will fail on Vault rebased patch